platforms/evm/protocols/portico/src/bridge.ts

```diff
@@ -349,4 +350,87 @@ export class EvmPorticoBridge<
     }
     return portico.uniswapQuoterV2;
   }
+
+  async getTransferResult(
+    vaa: PorticoBridge.VAA,
+  ): Promise<PorticoBridge.TransferResult> {
+    // First check if the transfer is completed
+    const isCompleted = await this.isTransferCompleted(vaa);
+    if (!isCompleted) return { swapResult: 'pending' };
+
+    const finalToken = Wormhole.tokenId(
+      this.chain,
+      vaa.payload.payload.flagSet.flags.shouldUnwrapNative
+        ? 'native'
+        : vaa.payload.payload.finalTokenAddress.toNative(this.chain).toString(),
+    );
+
+    const finalAmount = (() => {
+      const amountLessFee =
+        vaa.payload.payload.minAmountFinish - vaa.payload.payload.relayerFee;
+      return amountLessFee < 0n ? 0n : amountLessFee;
+    })();
+
+    const defaultResult: PorticoBridge.TransferResult = {
+      swapResult: 'success',
+      receivedToken: {
+        token: finalToken,
+        amount: finalAmount,
+      },
+    };
+
+    // This is a simplification since there is no swap on Ethereum
+    // since the highway token originates there
+    if (this.chain === 'Ethereum') return defaultResult;
+
+    // Check if the swap succeeded or failed
+    // NOTE: If we can't find the event, assume the swap succeeded
+    const tokenBridge = this.tokenBridge.tokenBridge;
+    const filter = tokenBridge.filters.TransferRedeemed(
+      toChainId(vaa.emitterChain),
+      vaa.emitterAddress.toString(),
+      vaa.sequence,
+    );
+
+    // Search for the event in the last 15 minutes
+    const latestBlock = await EvmPlatform.getLatestBlock(this.provider);
```

Comment:
We have to constrain the range we're searching for otherwise the RPC may return an error